dds/DCPS/WriterInfo.cpp

```diff
@@ -89,22 +93,22 @@ WriterInfo::WriterInfo(const WriterInfoListener_rch& reader,
 WriterInfo::~WriterInfo()
 {
   historic_samples_sweeper_task_->cancel();
+  liveliness_check_task_->cancel();
 }
 
-const char* WriterInfo::get_state_str() const
+const char* get_state_str(WriterState state)
 {
-  ACE_Guard<ACE_Thread_Mutex> guard(mutex_);
-  switch (state_) {
+  switch (state) {
   case NOT_SET:
     return "NOT_SET";
   case ALIVE:
     return "ALIVE";
   case DEAD:
     return "DEAD";
   default:
-    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: WriterInfo::get_state_str: ")
+    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR:get_state_str: ")
```

Comment:
```suggestion
    ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: get_state_str(WriterState): "
```

It would be good to have more context